### ISSUE TITLE
Directory being moved out of a watch dir don't get removed from the database

### ISSUE THREAD

**Issue by cjlucas** [cjlucas/archivedb#5]
```2011-11-12 11:51:59 :: DEBUG    :: monitor.py:223 :: process_IN_MOVED_FROM :: <Event cookie=33843 dir=True mask=0x40000040 maskname=IN_MOVED_FROM|IN_ISDIR name=Lynda.com path=/mnt/user/stuff/apps/cbt pathname=/mnt/user/stuff/apps/cbt/Lynda.com wd=76 >
2011-11-12 11:53:00 :: DEBUG    :: monitor.py:202 :: process_IN_CLOSE_WRITE :: <Event dir=False mask=0x8 maskname=IN_CLOSE_WRITE name=trigger.txt path=/mnt/user/stuff/movies pathname=/mnt/user/stuff/movies/trigger.txt wd=2661 >
2011-11-12 11:53:00 :: DEBUG    :: monitor.py:178 :: check_last_moved_from :: self.last_moved_from = /mnt/user/stuff/apps/cbt/Lynda.com
2011-11-12 11:53:00 :: DEBUG    :: monitor.py:197 :: check_last_moved_from :: it is assumed file was moved outside watch_dirs, deleting.
2011-11-12 11:53:00 :: DEBUG    :: sql.py:207 :: get_fields :: SELECT`id`FROM`archive` WHERE
        watch_dir = '/mnt/user/stuff/apps' and path = 'cbt' and filename = 'Lynda.com'
2011-11-12 11:53:00 :: DEBUG    :: monitor.py:105 :: delete_file :: file '/mnt/user/stuff/apps/cbt/Lynda.com' not found in database

```

As you can see, the program treats Lynda.com as a file rather than the directory that it is.
```

**Comment by cjlucas** [comment]
Note to self: I was only able to reproduce this behavior if the directory being moved had many subdirectories within subdirectories. Otherwise pyinotify would just call process_IN_DELETE on the file within the directory.

Example:

```2011-11-17 16:30:11 :: DEBUG    :: monitor.py:199 :: process_IN_CLOSE_WRITE :: <Event dir=False mask=0x8 maskname=IN_CLOSE_WRITE name=empty.txt path=/mnt/user/stuff/apps/cbt/testing pathname=/mnt/user/stuff/apps/cbt/testing/empty.txt wd=4642 >
2011-11-17 16:30:11 :: DEBUG    :: sql.py:221 :: get_fields :: SELECT`mtime`,`size`FROM`archive`WHERE
        watch_dir = '/mnt/user/stuff/apps' and path = 'cbt/testing' and filename = 'empty.txt'
2011-11-17 16:30:12 :: INFO     :: monitor.py:66 :: add_file :: generating checksum for empty.txt ...
2011-11-17 16:30:12 :: INFO     :: monitor.py:70 :: add_file :: inserting empty.txt into the database.
2011-11-17 16:30:12 :: DEBUG    :: sql.py:98 :: insert_file :: INSERT INTO`archive`
                (watch_dir, path, filename, md5, mtime, size) VALUES
                ('/mnt/user/stuff/apps', 'cbt/testing', 'empty.txt', 'd41d8cd98f00b204e9800998ecf8427e', '1321572611.0', '0')

``````
This is where I move the directory out of the watch dir:

```2011-11-17 16:30:18 :: DEBUG    :: monitor.py:209 :: process_IN_DELETE :: <Event dir=False mask=0x200 maskname=IN_DELETE name=empty.txt path=/mnt/user/stuff/apps/cbt/testing pathname=/mnt/user/stuff/apps/cbt/testing/empty.txt wd=4642 >
2011-11-17 16:30:18 :: DEBUG    :: sql.py:221 :: get_fields :: SELECT `id` FROM `archive` WHERE
        watch_dir = '/mnt/user/stuff/apps' and path = 'cbt/testing' and filename = 'empty.txt'
2011-11-17 16:30:18 :: INFO     :: monitor.py:102 :: delete_file :: removing empty.txt from the database.
2011-11-17 16:30:18 :: DEBUG    :: sql.py:152 :: delete_file :: DELETE FROM `archive` WHERE id = '52005'
2011-11-17 16:30:18 :: DEBUG    :: monitor.py:209 :: process_IN_DELETE :: <Event dir=True mask=0x40000200 maskname=IN_DELETE|IN_ISDIR name=testing path=/mnt/user/stuff/apps/cbt pathname=/mnt/user/stuff/apps/cbt/testing wd=78 >```
``````

**Comment by cjlucas** [comment]
When first coding this feature, I handled the 2nd example. Now the original example should be handled properly as well.

**Comment by cjlucas** [comment]
Will be included in v0.2.2.
